### ISSUE TITLE
Incremental Analysis: Detecting changed start states does not work when Hashconsing is enabled.

### ISSUE THREAD

**Issue by jerhard** [goblint/analyzer#361]
TD3 does not detect that a start state changed for a function when hashconsing is activated. 

Consider the following example code. After the first analysis, the initial value of the global is changed in the source code. If the example is incrementally analyzed with hashconsing, it is not the detected that the start state of `main` changed, so no reanalysis is triggered.
```
int g = 1; // Change the initial value to e.g. 5
int main(){
	int x = g;
	return x;
}
```
This has to do with the fact that with hashconsing activated only tags are compared that are erroneously duplicated.

**Comment by vogler** [comment]
Do you know the reason for the duplicate tags?
There's already an `equal_debug` in `Printable.HConsed`:
https://github.com/goblint/analyzer/blob/b517b410273a705f79f019389efd0339213c0c49/src/domains/printable.ml#L131-L144
